### PR TITLE
Adds Greytide implant

### PR DESCRIPTION
**Pull request by CMOisLing**
A cheaper mindslave implant (role restricted to assistants) that instead compels the subject to break into departments. They will still be loyal to their master and fellow assistants but wont murder or kill anyone. They are instead meant to serve as a distraction for sec or someone to help you break into places you don't have access for. I'm open to rewrites since it is difficult to get the wording right so people don't take the objective as "go around and kill sec" because thats not the intention.

:cl:  
rscadd: Adds Greytide implant for assistants
/:cl:

### CODE REVIEW

**Comment by CMOisLing** [comment]
Pacifism Yes or No?

**Comment by Sir-Lagsalot** [comment]
Traitor stuff that lets you do things without murder is an automatic 👍 from me, now we just need people to actually use it ;-;

**Comment by Alexandria404** [comment]
I dislike the reference to the term greytide IC.
Also in the description the person implanted it says you feel the urge to greytide, which is not an explenation anyone will understand.

tldr: remove injoke, make description clear

**Comment by CMOisLing** [comment]
> I dislike the reference to the term greytide IC.
> Also in the description the person implanted it says you feel the urge to greytide, which is not an explenation anyone will understand.
> 
> tldr: remove injoke, make description clear

We already have references to greytide in-game, but I will remove it as an explanation since everyones idea of greytide is different.

File: yogstation/code/modules/antagonists/traitor/datum_mindslave.dm [hunk 1]
```diff
@@ -11,4 +11,19 @@
...
+	show_in_antagpanel = FALSE
+	antag_moodlet = /datum/mood_event/focused
+	can_hijack = FALSE
+	var/mob/living/carbon/master // I think I need to keep this for log purposes
+
+/datum/objective/greytide
+	name = "Mindslave Objective" // alt "Take what you want, the station is yours. Avoid needless harm you are still a crewmemeber."
+	martyr_compatible = TRUE
+	completed = TRUE
+	explanation_text = "Break things, take what you want, avoid harming anyone doing so. Remain loyal to your master and assistant brothers."
```
Comment (alexkar598) [inline yogstation/code/modules/antagonists/traitor/datum_mindslave.dm:33]: You define it here but define it later when generating the objective

File: code/game/objects/items/implants/implant_greytide.dm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		return FALSE
+
+	to_chat(target, "<span class='userdanger'><FONT size = 3>You feel a strong urge to break everything.  You feel a strong loyalty to [holder.real_name] and your assistant brothers. You want to break into everything, but harming others isnt something you will do.</FONT></span>")
+
+	var/datum/antagonist/greytide/GT = new
+	target.mind.add_antag_datum(GT)
+	GT.master = user
+	var/datum/objective/greytide/new_objective = new /datum/objective/greytide
+	GT.objectives += new_objective
+	new_objective.explanation_text = "Never betray [holder.real_name] or abandon your assistant brothers!"
```
Comment (alexkar598) [inline code/game/objects/items/implants/implant_greytide.dm]: Make it clearer the person cannot do harm

File: code/game/objects/items/implants/implant_greytide.dm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	ADD_TRAIT(target, TRAIT_PACIFISM, "Greytide Implant")
+
+	log_game("[holder.ckey] enslaved [target.ckey] with a greytide implant")
```
Comment (alexkar598) [inline code/game/objects/items/implants/implant_greytide.dm:50]: message_admins it too

File: yogstation/code/game/objects/items/storage/uplink_kits.dm [hunk 4]
```diff
@@ -15,4 +15,10 @@
 	name = "Mindslave Implant (with injector)"
 
 /obj/item/storage/box/syndie_kit/imp_mindslave/PopulateContents()
-	new /obj/item/implanter/mindslave(src)
\ No newline at end of file
+	new /obj/item/implanter/mindslave(src)
+
+/obj/item/storage/box/syndie_kit/imp_greytide
+	name = "Greytide Implant (with injector)"
```
Comment (alexkar598) [inline yogstation/code/game/objects/items/storage/uplink_kits.dm:21]:
```suggestion
	name = "Greytide Implant Injector"
```

File: yogstation/code/modules/antagonists/traitor/datum_mindslave.dm [hunk 5]
```diff
@@ -11,4 +11,19 @@
...
\ No newline at end of file
+	explanation_text = "Serve your master no matter what!"
+
+/datum/antagonist/greytide
+	name = "Greytider"
+	antagpanel_category = "Mindslave"
+	job_rank = ROLE_TRAITOR
+	show_in_antagpanel = FALSE
+	antag_moodlet = /datum/mood_event/focused
+	can_hijack = FALSE
```
Comment (alexkar598) [inline yogstation/code/modules/antagonists/traitor/datum_mindslave.dm]: I'd say remove this line so that they don't count against traitor hijacks

File: code/game/objects/items/implants/implant_greytide.dm [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		if(!source.mind.has_antag_datum(/datum/antagonist))
+			to_chat(source,"<span class='notice'>You are no longer an antagonist.</span>")
+	REMOVE_TRAIT(source, TRAIT_PACIFISM, "Greytide Implant")
+
+/obj/item/implanter/greytide
+	name = "implanter (greytide)"
+
+/obj/item/implanter/greytide/Initialize()
+	imp = new /obj/item/implant/greytide(src)
+	..()
```
Comment (alexkar598) [inline code/game/objects/items/implants/implant_greytide.dm]: move this to the top of the proc but change it to . = ..()

File: code/game/objects/items/implants/implant_greytide.dm [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+	GT.objectives += new_objective
+	new_objective.explanation_text = "Never betray [holder.real_name] or abandon your assistant brothers!"
+	ADD_TRAIT(target, TRAIT_PACIFISM, "Greytide Implant")
+
+	log_game("[holder.ckey] enslaved [target.ckey] with a greytide implant")
+
+	return ..()
+
+/obj/item/implant/greytide/removed(mob/source)
+	if(!..())
```
Comment (alexkar598) [inline code/game/objects/items/implants/implant_greytide.dm]:
```suggestion
	. = ..()
	if(!.)
```

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: yogstation/code/game/objects/items/storage/uplink_kits.dm (hunk 4 above)
Comment (Lostlarry) [reply yogstation/code/game/objects/items/storage/uplink_kits.dm]: thats a box and mindslave is doing the same thing 6 lines above that

Comment (Lostlarry) [reply yogstation/code/game/objects/items/storage/uplink_kits.dm:21]: thats a box and mindslave is doing the same thing 6 lines above that

File: yogstation/code/modules/antagonists/traitor/datum_mindslave.dm (hunk 5 above)
Comment (Lostlarry) [reply yogstation/code/modules/antagonists/traitor/datum_mindslave.dm]: 0  is HIJACK_NEUTRAL which means ignored for purposes of hijack

Comment (alexkar598) [reply yogstation/code/modules/antagonists/traitor/datum_mindslave.dm]: Well this isn't a boolean anyways, just remove the line so it can go back to the default which is HIJACK_NEUTRAL

File: yogstation/code/modules/antagonists/traitor/datum_mindslave.dm (hunk 1 above)
Comment (CMOisLing) [reply yogstation/code/modules/antagonists/traitor/datum_mindslave.dm:33]: Don't I need to keep the explanation text in the implant so I can add the masters real name to the  text. And keep it in the mindslave datum so its defined.

**Review by alexkar598 — CHANGES_REQUESTED** [review]
Add a check on the antag datum on_clone to see if the clone has the implant and to clear itself out if not
